### PR TITLE
factor out the shared header of externs generation

### PR DESCRIPTION
**Pull request by evmar**
Rather than having annotate() return externs with a header (which means multiple
calls to annotate() can't be concatenated), instead expose an EXTERNS_HEADER
constant.

Use this in the tests (causing no test change).

Also fix the tests so that if a test *did* have multiple externs files, it wouldn't cause
a duplicated header.

### CODE REVIEW

File: src/tsickle.ts [hunk 1]
```diff
@@ -34,6 +34,18 @@ export interface Output {
 }
 
 /**
+ * The header to be used in generated externs.  This is not included in the
+ * output of annotate() because annotate() works one file at a time, and
+ * typically you create one externs file from the entire compilation.
```
Comment (mprobst) [inline src/tsickle.ts]: nit: entire compilation unit, as in one bazel target, right?

File: src/tsickle_compiler_host.ts [hunk 2]
```diff
@@ -6,7 +6,8 @@ import {convertDecorators} from './decorator-annotator';
 import {processES5} from './es5processor';
 import {ModulesManifest} from './modules_manifest';
 import * as sourceMapUtils from './source_map_utils';
-import {annotate, isDtsFileName} from './tsickle';
+import * as tsickle from './tsickle';
+import {isDtsFileName} from './tsickle';
```
Comment (mprobst) [inline src/tsickle_compiler_host.ts]: nit: use `tsickle.isDtsFileName` for consistency?

Comment (evmar) [inline src/tsickle_compiler_host.ts]: Yeah I struggled on this one.  I figured isDtsFileName itself is a pretty specific name (it doesn't even really belong in the tsickle module maybe?) but I guess it's trivial to change it later so I'll fix it.

Comment (mprobst) [inline src/tsickle_compiler_host.ts]: Ack, makes sense.

**Pull request merged**